o1vm/src/pickles/proof.rs

```diff
@@ -0,0 +1,27 @@
+use ark_ff::Zero;
+use kimchi::curve::KimchiCurve;
+
+pub struct WitnessColumns<G> {
+    // FIXME: add selectors
+    pub scratch: [G; crate::interpreters::mips::witness::SCRATCH_SIZE],
+    pub instruction_counter: G,
+    pub error: G,
+}
+
+pub struct ProofInputs<G: KimchiCurve> {
```

Comment:
I'd call that extended witness
There often is an ambiguity between the witness as the necessarry info to craft a proof (eg. a pre image, or the witness in a more complexity theory sense) and the witness columns which include all the intermediary variables of the constraint system.
I'd like to avoid that ambiguity, and proof inputs is ambigious, or makes me think of the necessarry info